tutorials/plot_receptive_field.py

```diff
@@ -0,0 +1,228 @@
+       The Multivariate Temporal Response Function (mTRF) Toolbox:
+       A MATLAB Toolbox for Relating Neural Signals to Continuous Stimuli.
+       Frontiers in Human Neuroscience 10, 604.
+       doi:10.3389/fnhum.2016.00604
+"""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#
+# License: BSD (3-clause)
+
+import mne
+import numpy as np
+import matplotlib.pyplot as plt
+
+from scipy.stats import multivariate_normal
+from scipy.io import loadmat
+from mne.decoding import ReceptiveField, delay_time_series
+from sklearn.preprocessing import scale
+from sklearn.linear_model import Ridge
+from sklearn.metrics import r2_score
+rng = np.random.RandomState(1337)  # To make this example reproducible
+
+###############################################################################
+# Load audio data
+# ---------------
+#
+# We'll read in the audio data from [3]_ in order to simulate a response.
+
+# Read in audio that's been recorded in epochs.
+path_audio = mne.datasets.mtrf.data_path()
+data = loadmat(path_audio + '/speech_data.mat')
+audio = data['spectrogram'].T
+sfreq = float(data['Fs'].squeeze())
+n_decim = 2
+audio = mne.filter.resample(audio, 1, n_decim, 'auto')
+sfreq /= n_decim
+
+###############################################################################
+# Create a receptive field
+# ------------------------
+#
+# We'll simulate a linear receptive field for a theoretical neural signal. This
+# defines how the signal will respond to power in this receptive field space.
+n_freqs = 16
+tmin, tmax = -.4, 0.
+
+# To simulate the data we'll create explicit delays here
+delays_samp = np.arange(np.round(tmin * sfreq),
+                        np.round(tmax * sfreq) + 1).astype(int)
+delays_sec = delays_samp / sfreq
+freqs = np.logspace(2, np.log10(5000), n_freqs)
+grid = np.array(np.meshgrid(delays_sec, freqs))
+
+# We need data to be shaped as n_epochs, n_features, n_times, so swap axes here
+grid = grid.swapaxes(0, -1).swapaxes(0, 1)
+
+# Simulate a temporal receptive field with a Gabor filter
+means_high = [-.1, 1000]
+means_low = [-.15, 2000]
+cov = [[.0005, 0], [0, 200000]]
+gauss_high = multivariate_normal.pdf(grid, means_high, cov)
+gauss_low = -1 * multivariate_normal.pdf(grid, means_low, cov)
+weights = gauss_high + gauss_low  # Combine to create the "true" STRF
+
+fig, ax = plt.subplots()
+ax.pcolormesh(delays_sec, freqs, weights, cmap='RdBu_r')
+ax.set(title='Simulated STRF', xlabel='Time Lags (s)', ylabel='Frequency (Hz)')
+plt.setp(ax.get_xticklabels(), rotation=45)
+plt.autoscale(tight=True)
+mne.viz.tight_layout()
+
+
+###############################################################################
+# Simulate a neural response
+# --------------------------
+#
+# Using this receptive field, we'll create an artificial neural response to
+# a stimulus.
+
+# Reshape audio to split into epochs, then make epochs the first dimension.
+n_epochs, len_epoch = 30, 3
```

Comment:
how about `n_seconds` since that's technically what the variable is? or `n_times_sec`?